### PR TITLE
feat(sdk/go): implement pg SDK for TinyGo 

### PR DESCRIPTION
**Pull request by Patrick0308**
feat(sdk/go): implement pg SDK for TinyGo

### CODE REVIEW

**Comment by Patrick0308** [comment]
The code about rdbms is similar on different db sdk. @adamreese  Should we move rdbms code into a internal pkg?

**Comment by adamreese** [comment]
I'm not sure if there's much code we can share because the C types that are generated per API are distinct types.

File: sdk/go/Makefile [hunk 1]
```diff
@@ -30,6 +30,7 @@ build-examples: $(EXAMPLES_DIR)/tinygo-key-value/main.wasm
 build-examples: $(EXAMPLES_DIR)/tinygo-sqlite/main.wasm
 build-examples: $(EXAMPLES_DIR)/tinygo-llm/main.wasm
 build-examples: $(EXAMPLES_DIR)/tinygo-mysql/main.wasm
+# build-examples: $(EXAMPLES_DIR)/tinygo-pgsql/main.wasm
```
Comment (adamreese) [inline sdk/go/Makefile:33]: Why is this commented out?

File: examples/tinygo-outbound-pg/spin.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+spin_manifest_version = "1"
+authors = ["Patrick Jiang <patrickjiang0530@gmail.com>"]
+description = ""
+name = "outbound-pg-example"
+trigger = { type = "http", base = "/" }
+version = "0.1.0"
+
+[[component]]
+environment = { DB_URL = "mysql://spin:spin@127.0.0.1/spin_dev" }
```
Comment (adamreese) [inline examples/tinygo-outbound-pg/spin.toml:9]: This should reference a postgres db

File: examples/tinygo-outbound-pg/main.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	Name      string
+	Prey      *string // nullable field must be a pointer
+	IsFinicky bool
+}
+
+func init() {
+	spinhttp.Handle(func(w http.ResponseWriter, r *http.Request) {
+
+		// addr is the environment variable set in `spin.toml` that points to the
+		// address of the Mysql server.
```
Comment (adamreese) [inline examples/tinygo-outbound-pg/main.go:24]:
```suggestion
		// address of the Postgres server.
```

**Review by adamreese — CHANGES_REQUESTED** [review]

**Comment by Patrick0308** [comment]
I tested executing outbound pg sql with parameters that was failed. Error log is 
```
query failed: Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E42601), message: "syntax error at or near \",\"", detail: None, hint: None, position: Some(Original(27)), where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("scan.l"), line: Some(1180), routine: Some("scanner_yyerror") }) }
```
It's same on rust outbound pg sdk too.

**Review by adamreese — APPROVED** [review]
Huge thanks for more great work!

**Pull request merged**